cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleServiceTest.java

```diff
@@ -452,6 +472,111 @@ public void testCreateAppDetailsArchive() throws Exception {
     deleteNamespace("ns3");
   }
 
+  @Test(expected = ApplicationNotFoundException.class)
+  public void testDeployWithoutMarkingAppAsLatest() throws Exception {
+    setPromotionToProdFlag(true);
+    String appName = "application_not_marked_latest";
+    Id.Application appId = Id.Application.from(Id.Namespace.DEFAULT, appName);
+    Id.Artifact artifactId = Id.Artifact.from(Id.Namespace.DEFAULT, "appWithConfig", "1.0.0-SNAPSHOT");
+    addAppArtifact(artifactId, ConfigTestApp.class);
+    HttpResponse resp = deployWithoutMarkingLatest(
+        appId, new AppRequest<>(ArtifactSummary.from(artifactId.toArtifactId())));
+    Assert.assertEquals(200, resp.getResponseCode());
+
+    applicationLifecycleService.getLatestAppDetail(appId.toEntityId().getAppReference());
+    setPromotionToProdFlag(false);
+  }
+
+  @Test
+  public void testMarkAppsAsLatest() throws Exception {
+    setPromotionToProdFlag(true);
+    setLcmEditFlag(false);
+
+    String appName = "application_to_be_marked_latest";
+    Id.Application appId = Id.Application.from(Id.Namespace.DEFAULT, appName);
+    Id.Artifact artifactId = Id.Artifact.from(Id.Namespace.DEFAULT, "appWithConfig", "1.0.0-SNAPSHOT");
+    addAppArtifact(artifactId, ConfigTestApp.class);
+    HttpResponse resp = deployWithoutMarkingLatest(
+        appId, new AppRequest<>(ArtifactSummary.from(artifactId.toArtifactId())));
+    Assert.assertEquals(200, resp.getResponseCode());
+
+    applicationLifecycleService.markAppsAsLatest(
+        Namespace.DEFAULT.toEntityId(),
+        new MarkLatestAppsRequest(
+            Collections.singletonList(new AppVersion(appName, ApplicationId.DEFAULT_VERSION))));
+
+    ApplicationDetail appDetail = applicationLifecycleService.getLatestAppDetail(appId.toEntityId().getAppReference());
+    Assert.assertNotNull(appDetail);
+    Assert.assertEquals(appName, appDetail.getName());
+    Assert.assertEquals(ApplicationId.DEFAULT_VERSION, appDetail.getAppVersion());
+
+    setPromotionToProdFlag(false);
```

Comment:
don't do stateful things like this in unit tests. If this test fails above, it won't reset anything and will break all the other tests in this class.

Instead, set the "normal" state in a `@Before` method to ensure it is always set for each test. If a test requires a flag to be set to a different value, do it at the start of the test.

Reply:
that being said, I don't think there is any need to set this in the test, just make it default to enabled. We're not impacting any of the existing logic with these new APIs.

Reply:
done.

I have removed the promotionToProd flag as it's not necessary for internal APIs.